src/zenml/artifacts/load_directory_materializer.py

```diff
@@ -0,0 +1,80 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Only-load materializer for directories."""
+
+import os
+import tempfile
+from pathlib import Path
+from typing import Any, ClassVar, Tuple, Type
+
+from zenml.enums import ArtifactType
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+class LoaderDirectoryMaterializer(BaseMaterializer):
+    """Materializer to load directories from the artifact store.
+
+    This materializer is very special, since it do not implement save
+    logic at all. The save of the data to some URI inside the artifact store
+    shall happen outside and is in user's responsibility.
+
+    This materializer solely supports the `link_folder_as_artifact` function.
+    """
+
+    ASSOCIATED_TYPES: ClassVar[Tuple[Type[Any], ...]] = (Path,)
+    ASSOCIATED_ARTIFACT_TYPE: ClassVar[ArtifactType] = ArtifactType.DATA
+
+    def load(self, data_type: Type[Any]) -> Any:
+        """Copy the artifact files to a local temp directory.
+
+        Args:
+            data_type: Unused.
+
+        Returns:
+            Path to the local directory that contains the artifact files.
+        """
+        directory = tempfile.mkdtemp(prefix="zenml-artifact")
```

Comment:
I'm guessing we could add a check that this is indeed a directory, not a file? Otherwise the copying will fail/have unexpected behaviour.